BrowserKit/Sources/TabDataStore/TabDataStore.swift

```diff
@@ -56,13 +73,25 @@ public actor DefaultTabDataStore: TabDataStore {
         }
     }
 
+    public func fetchBackupWindowData(forID id: UUID) async -> WindowData? {
```

Comment:
This can be called from fetchWindowData in the failure case.
Also both methods are very similar, I think you can create a separate helper method to reduce the repetition.